theme.go

```diff
@@ -270,3 +271,66 @@ func ThemeBase16() *Theme {
 
 	return &t
 }
+
+// themeCatppuccin generates a theme based a Catppuccin flavor.
+func themeCatppuccin(flavor catppuccin.Flavour) *Theme {
+	t := ThemeBase().copy()
+
+	var (
+		base     = lipgloss.Color(flavor.Base().Hex)
+		text     = lipgloss.Color(flavor.Text().Hex)
+		subtext1 = lipgloss.Color(flavor.Subtext1().Hex)
+		overlay1 = lipgloss.Color(flavor.Overlay1().Hex)
+		green    = lipgloss.Color(flavor.Green().Hex)
+		red      = lipgloss.Color(flavor.Red().Hex)
+		pink     = lipgloss.Color(flavor.Pink().Hex)
+		mauve    = lipgloss.Color(flavor.Mauve().Hex)
+		cursor   = lipgloss.Color(flavor.Rosewater().Hex)
+	)
+
+	f := &t.Focused
+	f.Base.BorderForeground(subtext1)
+	f.Title.Foreground(mauve)
+	f.Description.Foreground(text)
+	f.ErrorIndicator.Foreground(red)
+	f.ErrorMessage.Foreground(red)
+	f.SelectSelector.Foreground(pink)
+	f.Option.Foreground(text)
+	f.MultiSelectSelector.Foreground(pink)
+	f.SelectedOption.Foreground(green)
+	f.SelectedPrefix.Foreground(green)
+	f.UnselectedPrefix.Foreground(text)
+	f.UnselectedOption.Foreground(text)
+	f.FocusedButton.Foreground(base).Background(pink)
+	f.Next = f.FocusedButton.Copy()
+	f.BlurredButton.Foreground(text).Background(base)
+
+	f.TextInput.Cursor.Foreground(cursor)
+	f.TextInput.Placeholder.Foreground(overlay1)
+	f.TextInput.Prompt.Foreground(pink)
+
+	t.Blurred = f.copy()
+	t.Blurred.Base.BorderStyle(lipgloss.HiddenBorder())
+
+	return &t
+}
+
+// ThemeCatppuccinLatte returns a new theme based on the Latte flavor of the Catppuccin color scheme.
+func ThemeCatppuccinLatte() *Theme {
+	return themeCatppuccin(catppuccin.Latte)
+}
+
+// ThemeCatppuccinFrappe returns a new theme based on the Frappé flavor of the Catppuccin color scheme.
+func ThemeCatppuccinFrappe() *Theme {
+	return themeCatppuccin(catppuccin.Frappe)
+}
+
+// ThemeCatppuccinMacchiato returns a new theme based on the Macchiato flavor of the Catppuccin color scheme.
+func ThemeCatppuccinMacchiato() *Theme {
+	return themeCatppuccin(catppuccin.Macchiato)
+}
+
+// ThemeCatppuccinMocha returns a new theme based on the Mocha flavor of the Catppuccin color scheme.
+func ThemeCatppuccinMocha() *Theme {
```

Comment:
I've added the 4 flavours of Catppuccin here, without the use of `lipgloss.AdaptiveColor` as we have a light and dark theme. 

However, there's a case to be made where instead of doing all 4, we can have one single `Catppuccin` theme that uses `lipgloss.AdaptiveColor` to switch between our light flavour (Latte) and our darkest flavour (Mocha) accordingly. 

Now that I'm typing this out, I think that might actually be a nicer approach

Reply:
Thanks so much for going with this approach! I really like it!